### DISCUSSION TITLE
angent plugin tomcat，How do I get the body data？

### DISCUSSION THREAD

**Discussion by bufx** [apache/skywalking#9285] [Q&A]
the angent package name：tomcat-7.x-8.x-plugin-8.2.0.jar
TomcatInvokeInterceptor.collectHttpParam() is the only method currently available, but there's no way to get the body，How do I get the body，The inpuStream can be obtained only once，The question stumped me。

**Comment by wu-sheng** [comment]
> The inpuStream can be obtained only once

That is limited by Java API, there is no way to bypass it. You have to find another way to get the body.

**Reply by bufx** [reply]
Yes, I know，One way to get the request in the MVC project is through an interceptor，Then take it out and pack it，Then I use the request wrapped by myself，However, this approach does not seem to work with SkyWalking Agents。

**Reply by wu-sheng** [reply]
This has nothing related to the agent actually, we doesn't care.
